qubesbuilder/plugins/chroot_archlinux/__init__.py

```diff
@@ -0,0 +1,149 @@
+# The Qubes OS Project, http://www.qubes-os.org
+#
+# Copyright (C) 2023 Frédéric Pierret (fepitre) <frederic@invisiblethingslab.com>
+#
+# This program is free software; you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation; either version 2 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License along
+# with this program. If not, see <https://www.gnu.org/licenses/>.
+#
+# SPDX-License-Identifier: GPL-3.0-or-later
+
+from qubesbuilder.config import Config
+from qubesbuilder.distribution import QubesDistribution
+from qubesbuilder.executors import ExecutorError
+from qubesbuilder.log import get_logger
+from qubesbuilder.pluginmanager import PluginManager
+from qubesbuilder.plugins import ArchlinuxDistributionPlugin
+from qubesbuilder.plugins.chroot import ChrootError, ChrootPlugin
+
+log = get_logger("chroot_archlinux")
+
+
+def get_archchroot_cmd(chroot_dir, pacman_conf, makepkg_conf, additional_packages=None):
+    cmd = [f"sudo jinja2 {pacman_conf} -o /usr/local/share/devtools/qubes-x86_64.conf"]
+    additional_packages = additional_packages or []
+    mkarchchroot_cmd = [
+        "sudo",
+        "mkarchroot",
+        "-C",
+        "/usr/local/share/devtools/qubes-x86_64.conf",
+        "-M",
+        str(makepkg_conf),
+        str(chroot_dir),
+        "base-devel",
+    ] + additional_packages
+
+    cmd += [
+        "sudo pacman-key --init",
```

Comment:
Legacy builder tries very hard to not save the private key, by mounting ramfs at the `/etc/pacman.d/gnupg/private-keys-v1.d`. This one should too (and possibly re-running init if needs to re-sign some keys on archlinux-keyring update). If one build would leak that key, it could potentially compromise subsequent builds of different components (if, additionally, perform MitM somewhere).
For template build, it would be even more important. See builder-archlinux/scripts/arch-chroot-lite.

Reply:
I'm falling back to what was done before, init keyring each time and maybe ensure cleaning first.